src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/memberdays/MemberDaysReport.java

```diff
@@ -126,9 +128,20 @@ private String getDatePart(String dateTimeValue) {
                 .filter(x -> x.getValue().getListedDate().split("T")[0]
                     .equals(listingData.getListingDate()))
                 .collect(Collectors.toList());
+        } else {
+            return dateListedTypeItems.stream()
+                .filter(x -> isHearingDateInRange(x.getValue().getListedDate(),
+                        listingData.getListingDateFrom(), listingData.getListingDateTo()))
+                .collect(Collectors.toList());
         }
+    }
+
+    private boolean isHearingDateInRange(String dateListed, String dateFrom, String dateTo) {
+        var listed = LocalDate.parse(dateListed.split("T")[0]);
+        var from = LocalDate.parse(dateFrom);
+        var to = LocalDate.parse(dateTo);
 
-        return dateListedTypeItems;
+        return listed.compareTo(from) >= 0 && listed.compareTo(to) <= 0;
```

Comment:
This isn't particularly readable. The LocalData API has methods for date comparison.

Reply:
fixed